x-pack/plugins/apm/server/routes/services/route.ts

```diff
@@ -264,13 +265,37 @@ const serviceMetadataDetailsRoute = createApmServerRoute({
       kuery: '',
     });
 
-    return getServiceMetadataDetails({
+    const serviceMetadataDetails = await getServiceMetadataDetails({
       serviceName,
       setup,
       searchAggregatedTransactions,
       start,
       end,
     });
+
+    if (serviceMetadataDetails?.container?.ids) {
+      const {
+        savedObjects: { client: savedObjectsClient },
+        elasticsearch: { client: esClient },
+      } = await context.core;
+
+      const indexName = await getMetricIndices({
+        infraPlugin: plugins.infra,
+        savedObjectsClient,
+      });
```

Comment:
For other es clients the index is defined as part of the client

```suggestion
      const infraMetricEsClient = await getInfraMetricEsClient({
        infraPlugin: plugins.infra,
        core: context.core
      });
```

And we'll call it like:

```ts
const response = await infraMetricEsClient.search<unknown, ResponseAggregations>({
  query: {...}
})
```

Reply:
as we discussed offline I will address this in a follow up PR